src/hitch.c

```diff
@@ -2474,7 +2475,7 @@ handle_mgt_rd(struct ev_loop *loop, ev_io *w, int revents)
 		backaddr->ai_family = wu.payload.addr.family;
 		backaddr->ai_addrlen = wu.payload.addr.len;
 		memcpy(backaddr->ai_addr, &wu.payload.addr.addr,
-		       sizeof(struct sockaddr_storage));
```

Comment:
Wouldn't this overflow if we had an IPv4 address that is refresh to an IPv6 address?

Reply:
`struct sockaddr_storage` is designed to be family agnostic, see [RFC 2553 Section 3.10](https://tools.ietf.org/html/rfc2553#section-3.10).

Reply:
Hi @klenan, the problem here is that we need to consider the storage of what is being copied into. `backaddr->ai_addr` is either a `struct sockaddr_in` or a `struct sockaddr_in6`, both significantly smaller than the size of a `struct sockaddr_storage`.

The attempted fix doesn't work either, which is what my previous comment refers to.

The proper fix is probably to change the backaddr definition, possibly by making it a `struct sockaddr_storage`, capable of holding both IPv6 and IPv4 sockaddrs.

Reply:
That, or adopting `struct suckaddr`...